### ISSUE TITLE
Scope caused `atomWithObservable` to be out of sync

### ISSUE THREAD

**Issue by tien** [jotaijs/jotai-scope#36]
# Example

The 2 counters below while using the same atom are completely different

```tsx
const counterSubject = interval(1000).pipe(map((i) => `#${i}`));
const counterAtom = atomWithObservable(() => counterSubject);

const Counter = () => {
  const [counter] = useAtom(counterAtom);
  return <div>count: {counter}</div>;
};

export default function App() {
  return (
    <div className="App">
      <h1>Jotai scope bug</h1>
      <Counter />
      <ScopeProvider atoms={[]}>
        <Suspense>
          <Counter />
        </Suspense>
      </ScopeProvider>
    </div>
  );
}
```

# Reproduction

[Code sandbox link](https://codesandbox.io/p/sandbox/jotai-scope-observable-bug-nql23c?file=%2Fsrc%2FApp.tsx%3A23%2C24&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clx7su5ng00062e6a9h9vwlje%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clx7su5ng00022e6ahz7ipoin%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clx7su5ng00032e6a3zqoqvax%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clx7su5ng00052e6a7h6lhsi8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clx7su5ng00022e6ahz7ipoin%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx7su5ng00012e6acxdazzpp%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clx87kprb00022e6b2l2q2yxk%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A23%252C%2522startColumn%2522%253A24%252C%2522endLineNumber%2522%253A23%252C%2522endColumn%2522%253A24%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clx7su5ng00022e6ahz7ipoin%2522%252C%2522activeTabId%2522%253A%2522clx87kprb00022e6b2l2q2yxk%2522%257D%252C%2522clx7su5ng00052e6a7h6lhsi8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx7su5ng00042e6a18e2ryxo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clx7su5ng00052e6a7h6lhsi8%2522%252C%2522activeTabId%2522%253A%2522clx7su5ng00042e6a18e2ryxo%2522%257D%252C%2522clx7su5ng00032e6a3zqoqvax%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clx7su5ng00032e6a3zqoqvax%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by dmaskasky** [comment]
Related: [#25](https://github.com/jotaijs/jotai-scope/issues/25)

There is currently no way to distinguish between a derived atom that references a scoped atom, and a derived atom that does not. jotai-scope has to copy every derived atom to intercept their getter and setter to address this. As you discovered, this copy can cause problems.

PR [#35](https://github.com/jotaijs/jotai-scope/pull/35) adds an api to not copy specified derived atoms. I would merge this but I think there's a better way.

PR [#2609](https://github.com/pmndrs/jotai/pull/2609) makes a change to the store api that removes the need to copy derived atoms at all. 2609 should merge in a few weeks.

Once 2609 lands I'll rewrite jotai-scope to address the issue you found and [#25](https://github.com/jotaijs/jotai-scope/issues/25).

**Comment by dmaskasky** [comment]
Update: Integrating with https://github.com/pmndrs/jotai/pull/2609 is a lot harder than I imagined. I need to spend more time to understand how to rewrite jotai-scope following these changes.

Also, it looks like an atomWithObservable refactor is in the works with https://github.com/pmndrs/jotai/pull/2631, but unfortunately it looks as though the issue is still present.

https://codesandbox.io/p/sandbox/atom-with-observable-infinite-mount-bug-fix-forked-253y84?file=%2Fsrc%2FApp.tsx%3A6%2C1-7%2C1

**Comment by dmaskasky** [comment]
Closing as Duplicate of https://github.com/jotaijs/jotai-scope/issues/25. The extra computation described in that issue causes atomWithObservable to get a new value. The current strategy of cloning the atom is the root issue that is causing this behavior.
